llvm/lib/Transforms/Yk/ModuleClone.cpp

```diff
@@ -0,0 +1,78 @@
+#include "llvm/Transforms/Yk/ModuleClone.h"
+#include "llvm/IR/BasicBlock.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/Instruction.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/LLVMContext.h"
+#include "llvm/IR/Module.h"
+#include "llvm/InitializePasses.h"
+#include "llvm/Linker/Linker.h"
+#include "llvm/Pass.h"
+#include "llvm/Transforms/Utils/Cloning.h"
+
+#define DEBUG_TYPE "yk-module-clone-pass"
+
+using namespace llvm;
+
+namespace llvm {
+void initializeYkModuleClonePass(PassRegistry &);
+} // namespace llvm
+
+namespace {
+struct YkModuleClone : public ModulePass {
+  static char ID;
+  const std::string ClonePrefix = "__yk_clone_";
+
+  YkModuleClone() : ModulePass(ID) {
+    initializeYkModuleClonePass(*PassRegistry::getPassRegistry());
+  }
+
+  void updateClonedFunctions(Module &M) {
+    for (llvm::Function &F : M) {
+      if (F.hasExternalLinkage() && F.isDeclaration()) {
+        continue;
+      }
+      F.setName(ClonePrefix + F.getName().str());
+    }
+  }
+
+  void updateClonedGlobals(Module &M) {
+    for (llvm::GlobalVariable &GV : M.globals()) {
+      std::string GlobalName = GV.getName().str();
+      if (GlobalName.find('.') == 0) {
+        continue; // This is likely not user-defined. Example: `.L.str`.
+      }
+      GV.setInitializer(nullptr); // Remove the initializer
+      GV.setLinkage(llvm::GlobalValue::ExternalLinkage); // Set external linkage
+    }
+  }
+
+  bool runOnModule(Module &M) override {
+    if (M.getName() == "src/perf/collect.c") {
```

Comment:
What's the story with this?

Reply:
I thought it was an issue, but I can't reproduce it anymore
removed :point_right: https://github.com/ykjit/ykllvm/pull/202/commits/d223b645c1eb821c53ff09123ad8e7d7eaa0aeae